java/dataset/src/main/java/org/apache/arrow/dataset/scanner/ScanOptions.java

```diff
@@ -49,24 +51,72 @@ public ScanOptions(String[] columns, long batchSize) {
   /**
    * Constructor.
    * @param batchSize Maximum row number of each returned {@link org.apache.arrow.vector.ipc.message.ArrowRecordBatch}
-   * @param columns (Optional) Projected columns. {@link Optional#empty()} for scanning all columns. Otherwise,
+   * @param columnsSubset (Optional) Projected columns. {@link Optional#empty()} for scanning all columns. Otherwise,
    *                Only columns present in the Array will be scanned.
    */
-  public ScanOptions(long batchSize, Optional<String[]> columns) {
-    Preconditions.checkNotNull(columns);
+  public ScanOptions(long batchSize, Optional<String[]> columnsSubset) {
+    Preconditions.checkNotNull(columnsSubset);
     this.batchSize = batchSize;
-    this.columns = columns;
+    this.columnsSubset = columnsSubset;
+    this.columnsProduceOrFilter = Optional.empty();
   }
 
   public ScanOptions(long batchSize) {
     this(batchSize, Optional.empty());
   }
 
-  public Optional<String[]> getColumns() {
-    return columns;
+  public Optional<String[]> getColumnsSubset() {
```

Comment:
I don't think there's a compelling reason to rename everything. I do think it is time to add proper docstrings.

Reply:
rollback